onnxscript/ir/passes/common/shape_inference.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Shape inference pass using onnx.shape_inference."""
+
+from __future__ import annotations
+
+__all__ = [
+    "ShapeInferencePass",
+    "infer_shapes",
+]
+
+import logging
+
+import onnx
+
+from onnxscript import ir
+
+logger = logging.getLogger(__name__)
+
+# Temporarily remove initializers larger than this size to keep model size down
+# for the onnx.shape_inference call because it needs to serialize the model
+_BIG_TENSOR_SIZE_LIMIT = 1000  # 1KB
```

Comment:
nit: should this be a parameter, with a default value, of a ShapeInferencePass? likely won't matter in near term, but somebody might want to experiment with it some time.

Reply:
Thanks- I suppose we can expose it as needed when there is a need to keep the public apis limited for now.